Improve Wiki English (#2)
MCSR-Ranked/Wiki: docs/gameplay/rng.md

# RNG Standardization

## Explanation
For a fair and competitive nature, MCSR Ranked standardizes some RNG (Random Number Generation) through the **RNG Seed**. As a result, all players in the match will experience the same loot rates in the same order.

> The RNG Seed is same as Overworld seed by default. In private room, you can use different RNG Seed.

Reviewer: ```suggestion
The RNG Seed is same as Overworld seed by default. In private room, you can use a different RNG Seed.
```

This document will describe what is standardized based on RNG Seed.

## Piglin Barters

> - Barter chances for Ender Pearls and Obsidian is increased

Reviewer: ```suggestion
- Barter chances for Ender Pearls and Obsidian are increased
```

- 6 Obsidian is guaranteed in the first 72 barters (8 Gold Blocks)
- 3 Ender Pearl trades are guaranteed in the first 72 barters (8 Gold Blocks)
- All players in the match have the same trades in the same order

## Entity Drop Loot

> - Blaze Rods rates from Blaze are standardized
> - Ender Pearls rates from Enderman are standardized

Reviewer: ```suggestion
- Blaze Rod rates from Blazes are standardized
- Ender Pearl rates from Endermen are standardized
```

- Iron Golems will always drop 4 Iron
- Food drops from animals and Hoglins are standardized

## Block Drop Loot
- Flint rates from Gravel are standardized

> - Sticks from rates Dead Bush are standardized

Reviewer: ```suggestion
- Sticks rates from Dead Bushes are standardized
```

- Ender Dragon will force perch after 2 minutes 50 seconds (in-game time) from the spawned time or the last perched time

## Mob Spawners
- Blaze spawning timings are standardized
- Blaze spawning positions in spawners are standardized
::: details Details about Spawner standardization logic
The mod tries to attempt up to 4 spots in the spawner. These 4 spots are the same for both runners no matter which spawner. If any fire, block, entity or yourself is blocking the spot, you will NOT get that blaze. The time for these 4 spots to be chosen is the same, and thus this is why the blaze x,y,z are exactly the same as well. The 4 spots the mod tries to spawn are vanilla style, so it will have a large chance closer to the center of spawner.
:::
- Magma Cube spawning positions and sizes are standardized

## Mob Behavior
- Monsters do not spawn in Bastion Remnants except built-in entities (Can be toggled in Private Room settings)
- Monsters do not spawn in Desert Temples
- Zombified Piglins avoid pathfinding into Bastion Remnants
- Ghasts do not spawn within 5 chunks of the player
- Drowned never spawn with a held Trident

> - Elder Guardians never afflict the player with Mining Fatigue

Reviewer: ```suggestion
- Elder Guardians never curse the player with Mining Fatigue
```
honestly idt ive ever heard the word "afflict" before

Author: what really
i think its good as afflict

- Phantoms spawn and cycle is standardized
- Rain/storm cycle is standardized
- Villager trade offers are standardized

## Player & Portal Behaviors
- Player spawn coordinates are standardized
- Initial Nether entry location is standardized by spawn coordinates. The Y level behaviour is standardized to 64.
::: details
Every portal is treated like you are building on Y64 in the overworld, however this does not guarantee Y64 in the Nether. If the Nether is buried from Y64-Y31, then your portal will be Y30. Regardless, the portal is standardized for X,Y,Z.
:::

> - First blind portal* Y height is never be more than 5 blocks lower than your Nether Y height, so you get caved much less

Reviewer: ```suggestion
- First blind portal* Y height is never more than 5 blocks lower than your Nether Y height, so you get caved much less
```
